app/components/upload-app/status/index.ts

```diff
@@ -0,0 +1,177 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { computed } from '@ember/object';
+import ENUMS from 'irene/enums';
+import SubmissionModel from 'irene/models/submission';
+import Store from '@ember-data/store';
+import { inject as service } from '@ember/service';
+import DS from 'ember-data';
+import { addObserver, removeObserver } from '@ember/object/observers';
+import RealtimeService from 'irene/services/realtime';
+import { task } from 'ember-concurrency';
+import IntlService from 'ember-intl/services/intl';
+
+export default class UploadAppStatusComponent extends Component {
+  @tracked showDropdown = false;
+  @tracked anchorRef?: HTMLElement | null;
+  
+
+  @tracked submissions: DS.AdapterPopulatedRecordArray<SubmissionModel> | null =
+    null;
+
+  @service declare store: Store;
+  @service declare realtime: RealtimeService;
+  @service declare uploadApp: any;
+  @service declare intl: IntlService;
+
+  constructor(owner: unknown, args: any) {
+    super(owner, args);
+    this.initialize();
+  }
+
+  @computed('submissions.@each.status')
+  get submissionList() {
+    return this.submissions ? this.submissions : [];
+  }
+
+  @computed('submissions.@each.status', 'uploadedApps.files.length')
+  get submissionNumbers() {
+    let obj = {
+      completedSubmissions: 0,
+      failedSubmissions: 0,
+      runningSubmissions: 0,
+      variant: 'indeterminate',
+      loaderColor: 'green-loader',
+    };
+    if (this.submissions) {
+      this.submissions.map((submission) => {
+        if (submission.status === ENUMS.SUBMISSION_STATUS.ANALYZING) {
+          obj.completedSubmissions = obj.completedSubmissions + 1;
+        } else if (
+          ENUMS.SUBMISSION_STATUS_OBJ.FAILED.includes(submission.status)
+        ) {
+          obj.failedSubmissions = obj.failedSubmissions + 1;
+        } else {
+          obj.runningSubmissions = obj.runningSubmissions + 1;
+        }
+      });
+    }
+    if (this.uploadedApps?.files?.length > 0) {
+      obj.runningSubmissions =
+        obj.runningSubmissions + this.uploadedApps.files.length;
+    }
+    obj.variant = obj.runningSubmissions > 0 ? 'indeterminate' : 'determinate';
+    obj.loaderColor = obj.failedSubmissions > 0 ? '' : 'green-loader';
+    return obj;
+  }
+
+  get uploadedApps() {
+    return this.uploadApp.uploadingApps;
+  }
+
+  @action
+  registerAnchorRef(element: HTMLElement | null) {
+    this.anchorRef = element;
+  }
+
+  @computed('submissionList.@each.createdOn')
+  get sortedSubmissions() {
+    return this.submissionList
+      ?.slice()
```

Comment:
why is `slice` required here ?

Reply:
Using .slice() to create a copy and then sorting the copy